packages/dashboard/src/app/views/myStories/index.js

```diff
@@ -67,23 +69,49 @@ function MyStories() {
       storiesOrderById,
       totalPages,
       totalStoriesByStatus,
+      getAuthors,
     })
   );
 
-  const { filter, page, search, sort, view, showStoriesWhileLoading } =
+  const { filter, page, search, sort, view, showStoriesWhileLoading, author } =
     useStoryView({
       filters: STORY_STATUSES,
       isLoading,
       totalPages,
     });
 
+  const { setQueriedAuthors } = author;
+  const queryAuthorsBySearch = useCallback(
+    (authorSearchTerm) => {
+      return getAuthors(authorSearchTerm).then((data) => {
+        const userData = data.map(({ id, name }) => ({
+          id,
+          name,
+        }));
+        setQueriedAuthors((exisitingUsers) => {
+          const exisitingUsersIds = exisitingUsers.map(({ id }) => id);
+          const newUsers = userData.filter(
+            (newUser) => !exisitingUsersIds.includes(newUser.id)
+          );
+          return [...exisitingUsers, ...newUsers];
+        });
+      });
+    },
+    [getAuthors, setQueriedAuthors]
+  );
+
+  useEffect(() => {
+    queryAuthorsBySearch();
+  }, [queryAuthorsBySearch]);
```

Comment:
This only runs once, right?

Reply:
yea queryAuthorsBySearch just has a state setter as a dependency I think